# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1206029563 14400
#      Thu Mar 20 12:12:43 2008 -0400
# Branch release-3-0-x
# Node ID 017b7c79bd3c3bcccb401525237e0c61b76effff
# Parent  b16ad51511aae3118ae3a3c26ee61152a380cb7c
Ensure k is an integer scalar in rotdim

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-03-20  David Bateman  <dbateman@free.fr>
+
+	* general/rotdim.m: Ensure k is an integer scale.
+ 
 2008-03-19  Emil Lucretiu  <emil@la.mine.nu>
 
 	* signal/sinetone.m: Ensure integral number of samples.
 
 2008-03-19  Michael D. Godfrey  <godfrey@isl.stanford.edu>
 
 	* plot/__go_draw_axes__.m: Additional correction for symbol codes.
 
diff --git a/scripts/general/rotdim.m b/scripts/general/rotdim.m
--- a/scripts/general/rotdim.m
+++ b/scripts/general/rotdim.m
@@ -53,18 +53,18 @@
 
 function y = rotdim (x, k, plane)
   
   if (nargin < 1 || nargin > 3)
     print_usage ();
   endif
 
   if (nargin > 1 && ! isempty(k))
-    if (imag (k) != 0 || fix (k) != k)
-      error ("rotdim: k must be an integer");
+    if (!isscalar (k) || imag (k) != 0 || fix (k) != k)
+      error ("rotdim: k must be an scalar integer");
     endif
   else
     k = 1;
   endif
 
   nd = ndims (x);
   sz = size (x);
   if (nargin < 3)
